src/main/java/com/gmail/goosius/siegewar/listeners/SiegeWarLoreListener.java

```diff
@@ -0,0 +1,125 @@
+import org.bukkit.inventory.EquipmentSlot;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.persistence.PersistentDataHolder;
+
+import java.util.Arrays;
+
+/**
+ *
+ * @author lexiccn
+ *
+ */
+public class SiegeWarLoreListener implements Listener {
+
+    @EventHandler(priority = EventPriority.MONITOR)
+    public void onSiegeStart(SiegeWarStartEvent event) {
+        if (!SiegeWarSettings.isSiegeLoreEnabled()) return;
+
+        //Set name, type, attacker, defender, start time
+        SiegeWarLoreUtil.bannerSiegeStart(event.getSiege());
+    }
+
+    @EventHandler(priority = EventPriority.MONITOR)
+    public void onSiegeEnd(SiegeEndEvent event) {
+        if (!SiegeWarSettings.isSiegeLoreEnabled()) return;
+
+        //Set outcome, winner, points, end time
+        SiegeWarLoreUtil.bannerSiegeEnd(event.getSiege());
+    }
+
+    @EventHandler
+    public void onInteractBanner(PlayerInteractEvent event) {
+        if (!SiegeWarSettings.isSiegeLoreEnabled()) return;
+        if (event.getHand() != EquipmentSlot.HAND) return;
+
+        PersistentDataHolder holder;
+
+        if (event.getClickedBlock() != null && event.getAction() == Action.RIGHT_CLICK_BLOCK) {
+            if (!SiegeWarLoreUtil.isLoreItem(event.getClickedBlock().getState(), "siege_banner")) return;
+            holder = (PersistentDataHolder) event.getClickedBlock().getState();
+        } else if (event.getItem() != null && event.getAction() == Action.RIGHT_CLICK_AIR) {
+            if (!SiegeWarLoreUtil.isLoreItem(event.getItem(), "siege_banner")) return;
+            holder = event.getItem().getItemMeta();
+        } else return;
+
+        SiegeWarLoreUtil.sendBannerChat(holder, event.getPlayer());
+    }
+
+    @EventHandler
+    public void onPrepareCraft(PrepareItemCraftEvent event) {
+        if (!SiegeWarSettings.isSiegeLoreEnabled()) return;
+
+        ItemStack resultItem = event.getInventory().getResult();
+        if (resultItem == null) return;
+        if (resultItem.getType() != Material.SHIELD) return;
+
+        ItemStack bannerItem = Arrays.stream(event.getInventory().getMatrix())
+                .filter(itemStack -> SiegeWarLoreUtil.isLoreItem(itemStack, "siege_banner"))
+                .findFirst().orElse(null);
+        if (bannerItem == null) return;
+
+        SiegeWarLoreUtil.setShieldStackFromHolder(resultItem, bannerItem.getItemMeta());
+    }
+
+    @EventHandler(ignoreCancelled = true)
+    public void onTryStartSiege(PreSiegeCampEvent event) {
+        if (!SiegeWarSettings.isSiegeLoreEnabled()) return;
+        if (SiegeWarLoreUtil.isLoreItem(event.getFlag().getState(), "siege_banner")) return;
+
+        event.setCancelled(true);
+        event.setCancellationMsg(Translation.of("siege_lore_error_banner_cannot_be_used"));
+    }
+
+    //LOW to run before TownyBuildEvent, requires data not included there, but must run before SW listens for it
+    @EventHandler(priority = EventPriority.LOW, ignoreCancelled = true)
```

Comment:
It would be nice if a javadoc was added so its obvious what this method is used for.

Reply:
Added a small doc to it to try to explain the purpose, not marking as resolved because unsure if that wording or formatting is clear enough

Reply:
Maybe just a bit more, to the effect of why we do this, (I believe you said before that this is done because a Banner Item doesn't transfer the PDC to the Banner Block.

Reply:
Added a line "This is necessary as Spigot/Minecraft do not copy PDC or Lore from items to blocks (and back)"